inventory_management_system_api/repositories/utils.py

```diff
@@ -117,3 +121,62 @@ def compute_breadcrumbs(breadcrumb_query_result: list, entity_id: str, collectio
             f"collection '{collection_name}'"
         )
     return BreadcrumbsGetSchema(trail=trail, full_trail=full_trail)
+
+
+def create_move_check_aggregation_pipeline(entity_id: str, destination_id: str, collection_name: str) -> list:
+    """
+    Returns an aggregate query for checking whether an entity has been requested to move to one of its own children
+
+    :param entity_id: ID of the entity being moved
+    :param destination_id: ID of the entity it is being moved to (i.e. the new parent_id)
+
+    :raises InvalidObjectIdError: If the given entity_id or destination_id is invalid
+    :return: The query to feed to the collection's aggregate method. The value of list(result) should
+             be passed to check_move_result below.
+    """
+    return [
+        {"$match": {"_id": CustomObjectId(destination_id)}},
+        {
+            "$graphLookup": {
+                "from": collection_name,
+                "startWith": "$parent_id",
+                "connectFromField": "parent_id",
+                "connectToField": "_id",
+                "as": "ancestors",
+                "depthField": "level",
+                # Stop if hit the entity itself, no need to check further
+                "restrictSearchWithMatch": {"_id": {"$ne": CustomObjectId(entity_id)}},
+            }
+        },
+        # The following ensures that just a list of the parents containing only the parent_id's are returned
+        # in order from the top level down
+        {
+            "$facet": {
+                # Keep only these parameters
+                "root": [{"$project": {"parent_id": 1}}],
+                "ancestors": [
+                    {"$unwind": "$ancestors"},
+                    {
+                        "$sort": {
+                            "ancestors.level": -1,
+                        },
+                    },
+                    {"$replaceRoot": {"newRoot": "$ancestors"}},
+                    {"$project": {"parent_id": 1}},
+                ],
+            }
+        },
+        {"$project": {"result": {"$concatArrays": ["$ancestors", "$root"]}}},
+    ]
+
+
+def check_move_result(move_parent_check_result: list) -> bool:
```

Comment:
Same as my [comment](https://github.com/ral-facilities/inventory-management-system-api/pull/160#discussion_r1471378480) in your other PR - I think it would read better if the method name starts with `is` given it returns a boolean value.